RefactoringsExperiments/src/main/java/org/jetbrains/research/extractMethodExperiments/haas/Candidate.java

```diff
@@ -46,7 +49,7 @@ private void calculateRemainder() {
         ApplicationManager.getApplication().invokeAndWait(() -> {
             WriteCommandAction.runWriteCommandAction(originalMethod.getProject(), () -> {
                 for (PsiStatement statement : statementList) {
-                    statement.getParent().deleteChildRange(statementList.get(0), statementList.get(statementList.size() - 1));
+                    PsiTreeUtil.getParentOfType(statement, PsiCodeBlock.class).deleteChildRange(statementList.get(0), statementList.get(statementList.size() - 1));
```

Comment:
does it work as expected? I remember that we had problems with getting a parent for inner statements of` if `and `for` statements

Reply:
btw, extract `PsiTreeUtil.getParentOfType(statement, PsiCodeBlock.class)` into a separate variable.
is it always an instance of `PsiCodeBlock`?